backend/app/models/answer.rb

```diff
@@ -7,6 +7,25 @@ class Answer < ApplicationRecord
 
   validates :user_answer, presence: true, if: :free_text?
   validates :question_id, presence: true, uniqueness: { scope: :answers_survey_id }
+  validate :minimum_one_option
+  validate :maximum_one_option
 
   delegate :free_text?, to: :question, allow_nil: true
+  delegate :single_choice?, to: :question, allow_nil: true
+
+  private
+
+  def minimum_one_option
+    return if free_text? || !options.empty?
```

Comment:
I think we can reduce code complexity if we remove the `return`
```
if !free_text? && options.empty?
     errors.add(:options, :at_least_one_option)
end
```
or 
```
errors.add(:options, :at_least_one_option) if !free_text? && options.empty?
```

Reply:
Makes sense!